Source/WTF/wtf/text/StringCommon.h

```diff
@@ -689,7 +689,10 @@ ALWAYS_INLINE const char16_t* find16NonASCII(std::span<const char16_t> data)
 }
 #endif
 
-template<typename CharacterType1, typename CharacterType2, std::enable_if_t<std::is_integral_v<CharacterType1> && std::is_integral_v<CharacterType2> && sizeof(CharacterType1) == sizeof(CharacterType2)>* = nullptr>
+template<typename T> concept IsFindableCharacter = std::is_integral_v<T> || std::same_as<T, LChar>;
+
+template<IsFindableCharacter CharacterType1, IsFindableCharacter CharacterType2>
+    requires (sizeof(CharacterType1) == sizeof(CharacterType2))
```

Comment:
This seems like it would be subtly wrong if we ever added utf-8 support. Should this be `requires (std::same_as< CharacterType1, CharacterType2>)`? Although, maybe you just wouldn't ever want to use this on a utf-8 string anyway since it seems like the wrong algorithm. It looks wrong for utf-16 too but it's probably necessary for compatibility.